src/main/java/uk/gov/hmcts/reform/hmc/data/HearingEntity.java

```diff
@@ -91,10 +92,22 @@ public Integer getLatestRequestVersion() {
      */
     public Optional<HearingResponseEntity> getHearingResponseForLatestRequest() {
         Integer latestRequestVersion = getLatestRequestVersion();
-        return hasHearingResponses() ? getHearingResponses().stream()
-            .filter(hearingResponseEntity -> hearingResponseEntity.getRequestVersion().equals(latestRequestVersion))
-            .max(Comparator.comparing(HearingResponseEntity::getRequestTimeStamp))
-            :  Optional.empty();
+        if (hasHearingResponses()) {
+            List<HearingResponseEntity> filteredResponses = getHearingResponses().stream()
+                    .filter(hearingResponseEntity -> null != hearingResponseEntity
+                        && null != hearingResponseEntity.getRequestVersion()
+                        && hearingResponseEntity.getRequestVersion().equals(latestRequestVersion))
+                    .collect(Collectors.toUnmodifiableList());
+            if (!filteredResponses.isEmpty()) {
+                if (filteredResponses.size() == 1) {
+                    return Optional.ofNullable(filteredResponses.get(0));
+                }  else {
+                    return filteredResponses.stream().max(
+                            Comparator.comparing(HearingResponseEntity::getRequestTimeStamp));
+                }
+            }
+        }
+        return Optional.empty();
```

Comment:
What scenario is causing the need for this change? This is being used in some fundamental existing areas so would like to understand the issue.

Reply:
Didn't seem to be getting a value if only 1 response was there. Also helped with debugging. Have reverted.